bin/update-release-version.php

```diff
@@ -0,0 +1,173 @@
+    get-version:        Print the current version number
+
+EOT;
+
+    exit(1);
+}
+
+function read_release_version(string $filename): array
+{
+    if (! is_file($filename)) {
+        throw new Exception(sprintf('File not found: "%s"', $filename));
+    }
+
+    $contents = file_get_contents($filename);
+
+    $versions = [];
+
+    if (! preg_match('/^#define PHP_MONGODB_VERSION "(.*)"$/m', $contents, $matches)) {
+        throw new Exception('Could not match PHP_MONGODB_VERSION');
+    }
+    $versions['version'] = $matches[1];
+
+    if (! preg_match('/^#define PHP_MONGODB_STABILITY "(.*)"$/m', $contents, $matches)) {
+        throw new Exception('Could not match PHP_MONGODB_STABILITY');
+    }
+    $versions['stability'] = $matches[1];
+
+    if (! preg_match('/^#define PHP_MONGODB_VERSION_DESC (\d+),(\d+),(\d+),(\d+)$/m', $contents, $matches)) {
+        throw new Exception('Could not match PHP_MONGODB_VERSION_DESC');
+    }
+    $versions['versionComponents'] = array_slice($matches, 1);
+
+    return $versions;
+}
+
+function write_release_version(string $filename, array $version): void
+{
+    if (! is_file($filename)) {
+        throw new Exception(sprintf('File not found: "%s"', $filename));
+    }
+
+    $contents = file_get_contents($filename);
+
+    $patterns = [
+        '/^#define PHP_MONGODB_VERSION "(.*)"$/m',
+        '/^#define PHP_MONGODB_STABILITY "(.*)"$/m',
+        '/^#define PHP_MONGODB_VERSION_DESC (\d+),(\d+),(\d+),(\d+)$/m',
+    ];
+
+    $replacements = [
+        sprintf('#define PHP_MONGODB_VERSION "%s"', $version['version']),
+        sprintf('#define PHP_MONGODB_STABILITY "%s"', $version['stability']),
+        sprintf('#define PHP_MONGODB_VERSION_DESC %s,%s,%s,%s', ...$version['versionComponents']),
+    ];
+
+    $contents = preg_replace($patterns, $replacements, $contents, -1, $count);
+    if ($count !== 3) {
+        throw new Exception('Could not properly replace contents in file');
+    }
+
+    file_put_contents($filename, $contents);
+}
+
+function get_version_string_from_components(array $versionComponents): string
+{
+    return implode('.', array_slice($versionComponents, 0, 3));
+}
+
+function get_stable_version(array $versions): array
+{
+    if (! preg_match('/^(\d+\.\d+\.\d+)/', $versions['version'], $matches)) {
+        throw new Exception(sprintf('Version "%s" is not in the expected format', $versions['version']));
+    }
+
+    $version = $matches[1];
+
+    $expectedVersionString = get_version_string_from_components($versions['versionComponents']);
+
+    if ($expectedVersionString !== $version) {
+        throw new Exception(sprintf('Version "%s" does not match version from components ("%s")', $version, $expectedVersionString));
+    }
+
+    $newVersions = [
+        'version' => $version,
+        'stability' => 'stable',
+        'versionComponents' => $versions['versionComponents'],
+    ];
+
+    // Increase build number
+    $newVersions['versionComponents'][3]++;
```

Comment:
This is the only place I see the build number for `PHP_MONGODB_VERSION_DESC` being modified. Is there somewhere else that resets it to zero for new releases? Or maybe I'm missing a place where it's initialized to zero.

I just want to confirm that this bit works as expected.

Reply:
It's increased here for `get_stable_version`, while `get_next_patch_version` and `get_next_minor_version` explicitly set it to 0.